src/iris/bin/sender.py

```diff
@@ -755,37 +756,53 @@ def render(message):
 
 def mark_message_as_sent(message):
     connection = db.engine.raw_connection()
+
     params = [
         message['destination'],
         message['mode_id'],
         message.get('template_id'),
-        message['subject'],
-        message['body'],
     ]
+
     if 'aggregated_ids' in message:
         sql = SENT_MESSAGE_BATCH_SQL % connection.escape(message['aggregated_ids'])
         params.append(message['batch_id'])
     else:
         sql = SENT_MESSAGE_SQL
         params.append(message['message_id'])
+
     cursor = connection.cursor()
     if not message['subject']:
         message['subject'] = ''
         logger.warn('Message id %s has blank subject', message.get('message_id', '?'))
+
+    try:
+        cursor.execute(sql, params)
+        connection.commit()
+    except DataError:
+        logger.exception('Failed updating message metadata status (message ID %s) (application %s)', message.get('message_id', '?'), message.get('application', '?'))
+
+    # Update subject and body separately, as they may fail and we don't necessarily care if they do
     if len(message['subject']) > 255:
         message['subject'] = message['subject'][:255]
-    if len(message['body']) > 65000:
+
+    if len(message['body']) > 10000:
```

Comment:
Would it be better to do this check right after https://github.com/linkedin/iris/blob/master/src/iris/bin/sender.py#L903? Then we can skip the whole `distributed_send_message` call.

Problem with doing it after `distributed_send_message` is the user might receive the message delivered but it would be missing from the DB.